evalml/data_checks/ts_parameters_data_check.py

```diff
@@ -1,64 +1,121 @@
 """Data check that checks whether the time series parameters are compatible with the data size."""
 from evalml.data_checks import DataCheck, DataCheckError, DataCheckMessageCode
+from evalml.problem_types import handle_problem_types
 from evalml.utils.gen_utils import (
-    are_ts_parameters_valid_for_split,
     contains_all_ts_parameters,
+    is_ts_valid_for_split,
 )
 
 
 class TimeSeriesParametersDataCheck(DataCheck):
-    """Checks whether the time series parameters are compatible with data splitting.
+    """Checks whether the time series parameters and target data are compatible with data splitting.
```

Comment:
I argue we keep `TimeSeriesParametersDataCheck` the same, along with `are_ts_parameters_valid_for_split`, and add a new data check/util method for checking the class imbalance for time series classification.

My reasoning is that although the classification imbalance check and split size vs window size check are related, they are conceptually checking different things and that maps nicer to independent data checks and methods. Plus, in the future, if we decide we need to augment/improve our implementation for checking imbalanced classes for time series we can do that independently of the implementation for the split-size vs window size check.